crates/federate/src/send.rs

```diff
@@ -146,3 +153,34 @@ impl SendRetryTask<'_> {
     Ok(())
   }
 }
+
+#[derive(Serialize, Deserialize, Debug)]
+struct DummyActivity {
+  id: Url,
+  actor: Url,
+  #[serde(flatten)]
+  other: HashMap<String, Value>,
+}
+
+#[async_trait::async_trait]
+impl ActivityHandler for DummyActivity {
+  type DataType = LemmyContext;
+
+  type Error = LemmyError;
+
+  fn id(&self) -> &Url {
+    &self.id
+  }
+
+  fn actor(&self) -> &Url {
+    &self.actor
+  }
+
+  async fn verify(&self, _context: &Data<Self::DataType>) -> LemmyResult<()> {
+    Ok(())
+  }
+
+  async fn receive(self, _context: &Data<LemmyContext>) -> LemmyResult<()> {
+    Ok(())
+  }
+}
```

Comment:
did you consider to instead of deserializing the value at all, wrapping it in a newtype and just implementing the `id()` and `actor()` functions directly on the wrapper around serde_json::Value?

Reply:
I tried something like this but extracting the urls from `Value` is very ugly as it returns Option and then needs to parse the url. id and actor dont return result so all of that needs to be unwrapped. In theory thats fine as we know the fields are there, but this way is much cleaner.